### PR TITLE
Improve Compile-time sequences docs

### PR DESCRIPTION
**Pull request by ntrel**
* Change `expression list` to `value sequence` (this matches https://dlang.org/spec/template.html#variadic-templates point 2).
* Change `list` to `sequence`.
* Describe value & symbol sequences.
* Add subheadings.
* Tweak examples.
* Mention `tupleof`.

### CODE REVIEW

**Comment by wilzbach** [comment]
@ntrel your fork seems to be out of date.

File: articles/ctarguments.dd [hunk 1]
```diff
@@ -1,33 +1,35 @@
 Ddoc
 
-$(D_S Compile-time Argument Lists,
+$(D_S $(TITLE)),
```
Comment (wilzbach) [inline articles/ctarguments.dd]: :+1:

File: articles/ctarguments.dd [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
     )
 
-    $(P For historical reasons those sometimes can be called tuples in documentation or compiler
-        internals but don't get confused : this doesn't have much in common with tuples that
+    $(P $(B Note:) For historical reasons these sequences can sometimes be called tuples in documentation or compiler
+        internals, but don't get confused - they don't have much in common with tuples that
         commonly exist in other languages. Sequences of values of different types that can be
-        returned from functions are provided by $(LINK2 phobos/std_typecons.html#.Tuple, std.typecons.Tuple).
-        Using term "tuple" to mean compile-time lists is discouraged to avoid confusion, and if encountered
+        returned from functions are provided by $(PHOBOS typecons, .Tuple, std.typecons.Tuple).
```
Comment (wilzbach) [inline articles/ctarguments.dd]: `$(REF Tuple, std, typecons)`

File: articles/ctarguments.dd [hunk 3]
```diff
@@ -158,25 +164,29 @@ $(H2 $(LNAME2 auto-expansion, Auto-expansion))
     $(P This will only print `int` during compilation because the last line gets rewritten
         as `alias Dummy = Print0!(int, double)`. If auto-expansion didn't happen,
         `AliasSeq!(int, double)` would be printed instead. This is an inherent part of
-        the language semantics for variadic lists, and thus also preserved by `AliasSeq`.
+        the language semantics for variadic sequences, and thus also preserved by `AliasSeq`.
     )
 
-$(H2 $(LNAME2 homogenous-lists, Homogenous lists))
+$(H2 $(LNAME2 homogenous-sequences, Homogenous sequences))
```
Comment (wilzbach) [inline articles/ctarguments.dd]: `LEGACY_LNAME2` to avoid breakage?

File: articles/ctarguments.dd [hunk 4]
```diff
@@ -188,35 +198,70 @@ $(H2 $(LNAME2 homogenous-lists, Homogenous lists))
...
-        // refer to underlying compiler-generated arguments
+        // 'T' is a type sequence of argument types.
+        // 'args' is an instance of T whose elements are the function parameters
+        static assert(is(typeof(args) == T));
+        args[0] = T[0].init;
     }
     ---
 
-    $(P It is possible to use expression lists with values of the same type to declare array literals:)
+    $(P $(B Note:) $(PHOBOS typecons, .Tuple, std.typecons.Tuple) wraps a type sequence instance,
```
Comment (wilzbach) [inline articles/ctarguments.dd]: `$(REF Tuple, std, typecons)`

File: tuple.dd [hunk 5]
```diff
@@ -3,7 +3,7 @@ Ddoc
 For documentation on tuples that are commonly present in other languages (anonymous runtime entity that groups
 different values) refer to $(LINK2 phobos/std_typecons.html#tuple, Phobos documentation on std.typecons.tuple)
 
-You may be also looking for documentation for built-in $(LINK2 ctarguments.html, compile-time argument lists)
+You may be also looking for documentation for built-in $(LINK2 ctarguments.html, compile-time sequences)
```
Comment (wilzbach) [inline tuple.dd]: While you are at it, you could update the linkt to `articles/ctarguments.html` (not required, there are redirects)

Comment (ntrel) [reply tuple.dd]: Haven't done this, but fixed the other points. Thanks for reviewing.

File: dlang.org.ddoc [hunk 6]
```diff
@@ -331,6 +331,7 @@ _=
 PC=$(TC p, $1, $+)
 _=
 
+$(COMMENT Should be used only when link text is not std.$1.$2 - use REF instead)
 PHOBOS=$(SPANC phobos, $(AHTTPS dlang.org/phobos/std_$1.html#$2, $(TAIL $+)))
```
Comment (wilzbach) [inline dlang.org.ddoc]: Let's simply get rid of this deprecated macro: https://github.com/dlang/dlang.org/pull/2195

**Comment by wilzbach** [comment]
(used the `git commit --amend` trick to restart DAutoTest)

File: dlang.org.ddoc (hunk 6 above)
Comment (ntrel) [reply dlang.org.ddoc]: OK, reverted.

**Comment by ntrel** [comment]
@wilzbach Tests pass now ;-) Also I removed the comment for the `PHOBOS` macro so as not to conflict with your pull.

**Pull request merged**